src/Language/PureScript/Publish.hs

```diff
@@ -212,6 +212,11 @@ extractGithub =
     | ".git" `isSuffixOf` str = take (length str - 4) str
     | otherwise = str
 
+  matchUrl :: String -> Maybe String
+  matchUrl str
+    | "git@github.com:" `isPrefixOf` str = stripPrefix "git@github.com:" str
+    | otherwise = stripPrefix "git://github.com/" str
+
```

Comment:
Although I guess it doesn't matter if you're about to switch to parsec or something anyway. I am fairly sure this is the only part of the codebase that deals with these kinds of URLs, too.